include/alpaka/atomic/AtomicGenericSycl.hpp

```diff
@@ -0,0 +1,323 @@
+        }
+    };
+
+    // Max.
+    //! The SYCL accelerator atomic operation.
+    template<typename T, typename THierarchy>
+    struct AtomicOp<AtomicMax, experimental::AtomicGenericSycl, T, THierarchy>
+    {
+        static_assert(std::is_integral_v<T> || std::is_floating_point_v<T>, "SYCL atomics do not support this type");
+
+        static auto atomicOp(experimental::AtomicGenericSycl const&, T* const addr, T const& value) -> T
+        {
+            return experimental::detail::callAtomicOp<THierarchy>(
+                addr,
+                [&value](auto& ref) { return ref.fetch_max(value); });
+        }
+    };
+
+    // Exch.
+    //! The SYCL accelerator atomic operation.
+    template<typename T, typename THierarchy>
+    struct AtomicOp<AtomicExch, experimental::AtomicGenericSycl, T, THierarchy>
+    {
+        static_assert(std::is_integral_v<T> || std::is_floating_point_v<T>, "SYCL atomics do not support this type");
+
+        static auto atomicOp(experimental::AtomicGenericSycl const&, T* const addr, T const& value) -> T
+        {
+            return experimental::detail::callAtomicOp<THierarchy>(
+                addr,
+                [&value](auto& ref) { return ref.exchange(value); });
+        }
+    };
+
+    namespace detail
+    {
+        template<typename TRef, typename T, typename TEval>
+        inline auto casWithCondition(T* const addr, TEval&& eval)
+        {
+            auto ref = TRef{*addr};
+
+            auto old_val = ref.load();
+            auto assumed = T{};
+
+            do
+            {
+                assumed = old_val;
+                const auto new_val = eval(old_val);
+                old_val = ref.compare_exchange_strong(assumed, new_val);
+            } while(assumed != old_val);
+
+
+            return old_val;
+        }
+    } // namespace detail
+
+    // Inc.
+    //! The SYCL accelerator atomic operation.
+    template<typename T, typename THierarchy>
+    struct AtomicOp<AtomicInc, experimental::AtomicGenericSycl, T, THierarchy>
+    {
+        static_assert(std::is_unsigned_v<T>, "atomicInc not supported for non-unsigned types");
+
+        static auto atomicOp(experimental::AtomicGenericSycl const&, T* const addr, T const& value) -> T
+        {
+            auto inc = [&value](auto old_val) { return (old_val >= value) ? static_cast<T>(0) : (old_val + 1u); };
+            if(auto ptr = get_global_ptr(addr); ptr != nullptr)
+                return detail::casWithCondition<experimental::detail::global_ref<T, THierarchy>>(addr, inc);
+            else
+                return detail::casWithCondition<experimental::detail::local_ref<T, THierarchy>>(addr, inc);
+        }
+    };
+
+    // Dec.
+    //! The SYCL accelerator atomic operation.
+    template<typename T, typename THierarchy>
+    struct AtomicOp<AtomicDec, experimental::AtomicGenericSycl, T, THierarchy>
+    {
+        static_assert(std::is_unsigned_v<T>, "atomicDec not supported for non-unsigned types");
```

Comment:
Try to avoid double negatives to improve readability, e.g.:
```suggestion
        static_assert(std::is_unsigned_v<T>, "atomicDec only supported for unsigned types");
```
Applies to more places here.